src/config/env.ts

```diff
@@ -1,103 +1,105 @@
-import type { ExtendedWindow, PagoPAEnvVars } from '@/types/common.types'
-
-export const PAGOPA_ENV = (window as unknown as ExtendedWindow).pagopa_env
-
-export const NODE_ENV = import.meta.env.MODE
-export const isDevelopment = !!(import.meta.env.MODE === 'development')
-export const isProduction = !!(import.meta.env.MODE === 'production')
-export const isTest = !!(import.meta.env.MODE === 'test')
-
-export const TEST_MIXPANEL_PROJECT_ID = import.meta.env.REACT_APP_TEST_MIXPANEL_PROJECT_ID
-
-if (!PAGOPA_ENV && !isTest) {
-  console.warn('pagopa_env not available.')
+import type { ExtendedWindow } from '@/types/common.types'
+import { z } from 'zod'
+
+const viteConfigMode = z.enum(['development', 'production', 'test'])
+type ViteConfigMode = z.infer<typeof viteConfigMode>
+
+const GeneralConfigs = z.object({
+  NODE_ENV: z.enum(['development', 'production', 'test']),
+  STAGE: z.enum(['DEV', 'PROD', 'UAT', 'ATT', 'QA', 'VAPT']),
+  MIXPANEL_PROJECT_ID: z.string(),
+  ONETRUST_DOMAIN_SCRIPT_ID: z.string(),
+  CLIENT_ASSERTION_JWT_AUDIENCE: z.string(),
+  WELL_KNOWN_URLS: z.string(),
+  PRODUCER_ALLOWED_ORIGINS: z.string(),
+  PUBLIC_URL: z.string().default('/ui'),
```

Comment:
The `/ui` is not really a config that the backend passes. 
More like a constant. Where it is used? I would start thinking getting rid of it